src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1703,11 +1703,26 @@ private static void ReportImplicitImplementationMatchDiagnostics(Symbol interfac
                 }
             }
 
-            if (implicitImpl.IsStatic && !implementingType.ContainingAssembly.RuntimeSupportsStaticAbstractMembersInInterfaces)
+            if (implicitImpl.IsStatic && interfaceMember.ContainingModule != implementingType.ContainingModule)
             {
-                diagnostics.Add(ErrorCode.ERR_RuntimeDoesNotSupportStaticAbstractMembersInInterfacesForMember,
-                                GetInterfaceLocation(interfaceMember, implementingType),
-                                implicitImpl, interfaceMember, implementingType);
+                LanguageVersion requiredVersion = MessageID.IDS_FeatureStaticAbstractMembersInInterfaces.RequiredVersion();
+                LanguageVersion? availableVersion = implementingType.DeclaringCompilation?.LanguageVersion;
```

Comment:
> In which scenarios does the implementing type lack a declaring compilation? Maybe only in EE?

Types not declared in source do not have declaring compilation

Reply:
>  I didn't realize that these checks could occur when the implementing type isn't declared in source. Could you please point out a test where the implementing type is not declared in source?

The ```FindImplementationForInterfaceMember``` API can be called on any type and it might potentially get here for a type that is not declared is source. I don't think I specifically test for a diagnostics like this in those cases because the diagnostics is simply getting dropped, even if reported. ```StaticAbstractMembersInInterfacesTests``` call ```FindImplementationForInterfaceMember``` API in various scenarios for source and non-source symbols, as well as other tests utilizing the API.

Reply:
That makes complete sense, thanks.